site/frontend/src/storage.ts

```diff
@@ -0,0 +1,48 @@
+import {ref, Ref, watch} from "vue";
+
+/**
+ * Value that is persisted in LocalStorage.
+ */
+class StoredValue<T> {
+  private _value: T;
+
+  constructor(private key: string, defaultValue: T) {
+    const found = localStorage.getItem(key);
```

Comment:
We may need to first feature-detect whether `localStorage` is both supported and available ? (e.g. browser availability + whether the user has enabled it + some of the usage quota -- maybe more I don't know).